### DISCUSSION TITLE
ACT URLs with improper values should return error http header codes not 200s

### DISCUSSION THREAD

**Discussion by nep** [ExpressionEngine/ExpressionEngine#4571] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
Accessing an ACT URL with an improper value (number or string) should generate a proper error response code header instead of a 200 response.  Here are some current examples and whether they are problematic or not:

https://hopstudios.com/?ACT=15  -- 403   (fine)
https://hopstudios.com/?ACT=15.0  -- 403   (weird but fine)
https://hopstudios.com/?ACT=1.5e1  -- 403   (weird but fine)
https://hopstudios.com/?ACT=6  -- 200   (displays message, problem)
https://hopstudios.com/?ACT=8  -- 200   (displays message, problem)
https://hopstudios.com/?ACT=15.1  -- 200   (displays nothing, problem)
https://hopstudios.com/?ACT=999  -- 200   (displays nothing, problem)
https://hopstudios.com/?ACT=sdf  -- 200   (displays nothing, problem)
https://hopstudios.com/?ACT=6.2e3  -- 200   (displays nothing, problem)

**Describe the solution you'd like**
I think a 400 would be best, but 406 might also be okay.

https://www.semrush.com/blog/http-status-codes/

**Describe alternatives you've considered**
 Some specific ACTs could use other values, but I think as a default, 400 is good.

**Teachability, Documentation, Adoption, Migration Strategy**
No documentation changes would be necessary.

**Comment by intoeetive** [comment]
https://github.com/ExpressionEngine/ExpressionEngine/pull/4619
